cmd/oras/root/backup.go

```
@@ -0,0 +1,423 @@
+	if err != nil {
+		return 0, fmt.Errorf("failed to count referrers for tag %q, digest %q: %w", tag, root.Digest.String(), err)
+	}
+	referrerCount := len(referrers)
+	if root.MediaType != ocispec.MediaTypeImageIndex && root.MediaType != docker.MediaTypeManifestList {
+		// If the root is not an image index or manifest list, we have counted all referrers
+		return referrerCount, nil
+	}
+
+	// count referrers of children manifests
+	manifestBytes, err := content.FetchAll(ctx, target, root)
+	if err != nil {
+		return 0, fmt.Errorf("failed to fetch content of tag %q, digest %q: %w", tag, root.Digest.String(), err)
+	}
+	var index ocispec.Index
+	if err = json.Unmarshal(manifestBytes, &index); err != nil {
+		return 0, fmt.Errorf("failed to unmarshal index for tag %q, digest %q: %w", tag, root.Digest.String(), err)
+	}
+	childrenReferrers, err := graph.RecursiveFindReferrers(ctx, target, index.Manifests, extCopyGraphOpts)
+	if err != nil {
+		return 0, fmt.Errorf("failed to count referrers for children manifests of tag %q, digest %q: %w", tag, root.Digest.String(), err)
+	}
+	referrerCount += len(childrenReferrers)
+	return referrerCount, nil
+}
+
+// finalizeBackupOutput finalizes the backup output by removing temporary directories and exporting to a tar archive if needed.
+func finalizeBackupOutput(dstRoot string, opts *backupOptions, logger logrus.FieldLogger, metadataHandler metadata.BackupHandler) (returnErr error) {
+	// Remove ingest dir for a cleaner output
+	ingestDir := filepath.Join(dstRoot, "ingest")
+	if err := os.RemoveAll(ingestDir); err != nil && !errors.Is(err, fs.ErrNotExist) {
+		logger.Debugf("failed to remove ingest directory: %v", err)
+	}
+	if opts.outputFormat != outputFormatTar {
+		// If output format is not a tar, we are done
+		return nil
+	}
+
+	// exporting the backup to a tar archive
+	if err := metadataHandler.OnTarExporting(opts.output); err != nil {
+		return err
+	}
+	tarFile, err := os.Create(opts.output)
```

Can we test whether `ots.output` is creatable before the actual backup?

---

so that we can fail fast as it is a common failure path.

---

Updated to test at the beginning